compiler/rustc_lint/src/types.rs

```diff
@@ -1144,10 +1144,6 @@ pub(crate) fn repr_nullable_ptr<'tcx>(
             [var_one, var_two] => match (&var_one.fields.raw[..], &var_two.fields.raw[..]) {
```

Comment:
Okay so regarding the FIXME above... way back in the day we had a `fn represent_type_uncached` in `src/librustc_trans/trans/adt.rs` that also had logic for "detecting an option". If I understand the code correctly, that is actually the code that does layout computation -- seems like that lived in trans back then. Meanwhile layout computation no longer special-cases `Option`, we have way more generic enum optimizations.

So, the "FIXME: This duplicates code in codegen" above should be removed.